cpp/src/arrow/ipc/reader.cc

```diff
@@ -108,6 +109,20 @@ Status InvalidMessageType(MessageType expected, MessageType actual) {
 // ----------------------------------------------------------------------
 // Record batch read path
 
+/// \brief Structure to keep common arguments to be passed
+struct IpcReadContext {
+  IpcReadContext(DictionaryMemo* memo, const IpcReadOptions& option, bool swap)
+      : dictionary_memo(memo), options(option), swap_endian(swap) {}
+
+  DictionaryMemo* dictionary_memo;
+
+  const IpcReadOptions& options;
+
+  /// \brief LoadRecordBatch() or LoadRecordBatchSubset() swaps endianness of elements
+  /// if this flag is true
+  const bool swap_endian;
+};
```

Comment:
Would it be useful to add `MetadataVersion` and `Compression::type` here?

Reply:
Most use cases of  `IpcReadContext`, `ReadRecordBatchInternal()` do not receieve `MetadataVersion` and `Compression::type`.

If we pass default values for these two variables at the constructor, it would be useful to add them. So, we do not need to pass values to `IpcReadContent ipc(...)` before calling ReadRecordBatchInternal().

Do you want to add them while we use default values at the constructor?  I am neutral on this.

Reply:
If they're not used most of the time, then no, no need for it.